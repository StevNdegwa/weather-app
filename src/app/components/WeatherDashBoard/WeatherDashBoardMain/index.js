import WeatherDashBoardMain from "./WeatherDashBoardMain";
export default WeatherDashBoardMain;
