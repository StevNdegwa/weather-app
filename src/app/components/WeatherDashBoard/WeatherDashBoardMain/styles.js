import { Stack, Typography, styled } from "@mui/material";

export const MainWrapper = styled(Stack)`
width: 100%;
height: 100%;
padding: 1rem;
display: flex;
align-items: center;
justify-content: center;
`;

export const CurrentLocation = styled(Typography)`
font-size: var(--font-size-lg);
font-weight: lighter;
`;
