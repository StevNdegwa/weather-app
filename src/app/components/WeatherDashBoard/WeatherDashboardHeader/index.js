import WeatherDashboardHeader from "./WeatherDashboardHeader";
export default WeatherDashboardHeader;