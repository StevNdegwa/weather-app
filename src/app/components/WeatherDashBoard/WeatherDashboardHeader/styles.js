import { styled, Box } from "@mui/material";

export const HeaderWrapper = styled(Box)`
width: 100%;
height: 60px;
display: flex;
justify-content: space-between;
align-items: center;
padding:0 1rem;
`;
