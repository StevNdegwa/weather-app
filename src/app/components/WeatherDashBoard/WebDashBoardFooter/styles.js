import { styled, Box } from "@mui/material";

export const FooterWrapper = styled(Box)`
width: 100%;
height: 70px;
display: flex;
justify-content: center;
align-items:center;
padding: 0 1rem;
`;