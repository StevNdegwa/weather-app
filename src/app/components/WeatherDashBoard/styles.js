import { Box, Stack, styled } from "@mui/material";

export const WeatherDashBoardWrapper = styled(Box)`
width: 100%;
height: 100%;
display: flex;
justify-content:center;
align-items: center;
`;

export const WeatherDashBoardContent = styled(Stack)`
max-width:800px;
width: 100%;
max-height: 600px;
height: 100%;
`;
