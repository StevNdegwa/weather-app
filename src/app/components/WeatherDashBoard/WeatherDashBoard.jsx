import { useCallback, useMemo, useRef, useState } from "react";
import { Fade } from "@mui/material";
import PropTypes from "prop-types";
import TemperatureChart from "../TemperatureChart";
import DataLoadingIndicator from "../DataLoadingIndicator";
import WeatherDashboardHeader from "./WeatherDashboardHeader";
import WebDashBoardFooter from "./WebDashBoardFooter";
import WeatherDashBoardMain from "./WeatherDashBoardMain";
import { WeatherDashBoardWrapper, WeatherDashBoardContent } from "./styles";
import getDayWeatherAverage from "../../utils/getDayWeatherAverage";

export default function WeatherDashBoard({
  fadeIn,
  forecast = [],
  loadingData,
  reloadData,
}) {
  const [showingTempChart, setShowingTempChart] = useState(false);
  const sliderRef = useRef();
  const openTempChart = useCallback(
    () => setShowingTempChart(true),
    [setShowingTempChart]
  );
  const closeTempChart = useCallback(
    () => setShowingTempChart(false),
    [setShowingTempChart]
  );

  const slideToHome = useCallback(() => {
    sliderRef.current.goTo(0);
  }, []);

  const dailyAvgWeather = useMemo(() => {
    return getDayWeatherAverage(forecast);
  }, [forecast]);

  return (
    <Fade in={fadeIn}>
      <WeatherDashBoardWrapper>
        <TemperatureChart open={showingTempChart} closeModal={closeTempChart} />
        <DataLoadingIndicator loading={loadingData} />
        <WeatherDashBoardContent>
          <WeatherDashboardHeader reloadData={reloadData} />
          <WeatherDashBoardMain
            ref={sliderRef}
            forecastData={dailyAvgWeather}
            openTempChart={openTempChart}
          />
          <WebDashBoardFooter slideToHome={slideToHome} />
        </WeatherDashBoardContent>
      </WeatherDashBoardWrapper>
    </Fade>
  );
}

WeatherDashBoard.propTypes = {
  fadeIn: PropTypes.bool.isRequired,
  loadingData: PropTypes.bool.isRequired,
  reloadData: PropTypes.func.isRequired,
  error: PropTypes.object,
  forecast: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.string,
      temperature: PropTypes.number,
    })
  ),
};
