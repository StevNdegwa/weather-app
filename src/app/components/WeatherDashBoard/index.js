import WeatherDashBoard from "./WeatherDashBoard";
export default WeatherDashBoard;