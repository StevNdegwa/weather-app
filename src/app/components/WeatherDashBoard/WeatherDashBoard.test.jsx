import { mount } from "enzyme";
import { Provider } from "react-redux";
import fetchMock from "jest-fetch-mock";
import { act } from "@testing-library/react";
import store from "../../features/store";
import WeatherDashBoard from "./WeatherDashBoard";
import WeatherForecastCard from "../WeatherForecastCard";
import TemperatureChart from "../TemperatureChart";
import { WeatherDashBoardWrapper } from "./styles";

describe("test <WeatherDashBoard/> component", () => {
  let wrapper,
    reloadData = jest.fn();

  beforeEach(() => {
    fetchMock.resetMocks();

    wrapper = mount(
      <Provider store={store}>
        <WeatherDashBoard
          fadeIn={true}
          forecast={[
            {
              date: "2021-11-29 21:00:00",
              temperature: 30.16,
            },
          ]}
          loadingData={false}
          reloadData={reloadData}
        />
      </Provider>
    );
  });

  test("snapshot testing", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should contain all components", () => {
    expect(wrapper.find(WeatherDashBoardWrapper)).toHaveLength(1);
    expect(wrapper.find(WeatherForecastCard).length).toBeGreaterThanOrEqual(1);
    expect(wrapper.find(TemperatureChart)).toHaveLength(1);
  });

  test("should show and hide temperature chart", async () => {
    expect(wrapper.find(TemperatureChart).prop("open")).toBe(false);

    await act(async () => {
      wrapper.find(WeatherForecastCard).first().prop("showTempChart")();
    });

    wrapper.update();

    expect(wrapper.find(TemperatureChart).prop("open")).toBe(true);

    await act(async () => {
      wrapper.find(TemperatureChart).prop("closeModal")();
    });
    
    wrapper.update();

    expect(wrapper.find(TemperatureChart).prop("open")).toEqual(false);
  });
});
