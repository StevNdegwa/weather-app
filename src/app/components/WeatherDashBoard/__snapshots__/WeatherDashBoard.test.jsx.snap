// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <WeatherDashBoard/> component snapshot testing 1`] = `"<div class=\\"MuiBox-root css-sajir2\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"><div class=\\"MuiBox-root css-f48abj\\" style=\\"opacity: 0; visibility: hidden;\\"><div aria-busy=\\"true\\" class=\\"\\"><svg width=\\"80\\" height=\\"80\\" viewBox=\\"0 0 105 105\\" fill=\\"#B9B9B9\\" aria-label=\\"audio-loading\\"><circle cx=\\"12.5\\" cy=\\"12.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"0s\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"12.5\\" cy=\\"52.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"100ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"52.5\\" cy=\\"12.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"300ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"52.5\\" cy=\\"52.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"600ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"92.5\\" cy=\\"12.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"800ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"92.5\\" cy=\\"52.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"400ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"12.5\\" cy=\\"92.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"700ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"52.5\\" cy=\\"92.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"500ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle><circle cx=\\"92.5\\" cy=\\"92.5\\" r=\\"12.5\\"><animate attributeName=\\"fill-opacity\\" begin=\\"200ms\\" dur=\\"1s\\" values=\\"1;.2;1\\" calcMode=\\"linear\\" repeatCount=\\"indefinite\\"></animate></circle></svg></div></div><div class=\\"css-1eaknel-MuiStack-root\\"><div class=\\"MuiBox-root css-2uf0n5\\"><fieldset class=\\"MuiFormControl-root css-1nrlq1o-MuiFormControl-root\\"><legend class=\\"MuiFormLabel-root MuiFormLabel-colorPrimary css-u4tvz2-MuiFormLabel-root\\">Temperature scale</legend><div class=\\"MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root\\" role=\\"radiogroup\\" aria-label=\\"scale\\"><label class=\\"MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root\\" aria-label=\\"Degrees celcius scale\\" tabindex=\\"1\\"><span class=\\"MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root Mui-checked css-vqmohf-MuiButtonBase-root-MuiRadio-root\\"><input class=\\"PrivateSwitchBase-input css-1m9pwf3\\" name=\\"temperature-scale\\" type=\\"radio\\" value=\\"C\\" checked=\\"\\"><span class=\\"css-hyxlzm\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"RadioButtonUncheckedIcon\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"RadioButtonCheckedIcon\\"><path d=\\"M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z\\"></path></svg></span><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></span><span class=\\"MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root\\">°C</span></label><label class=\\"MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root\\" aria-label=\\"Degrees farenheit scale\\" tabindex=\\"2\\"><span class=\\"MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-vqmohf-MuiButtonBase-root-MuiRadio-root\\"><input class=\\"PrivateSwitchBase-input css-1m9pwf3\\" name=\\"temperature-scale\\" type=\\"radio\\" value=\\"F\\"><span class=\\"css-hyxlzm\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"RadioButtonUncheckedIcon\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z\\"></path></svg><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" data-testid=\\"RadioButtonCheckedIcon\\"><path d=\\"M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z\\"></path></svg></span><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></span><span class=\\"MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root\\">°F</span></label></div></fieldset><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"refresh data\\" data-mui-internal-clone-element=\\"true\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M20,8 C18.5974037,5.04031171 15.536972,3 12,3 C7.02943725,3 3,7.02943725 3,12 C3,16.9705627 7.02943725,21 12,21 L12,21 C16.9705627,21 21,16.9705627 21,12 M21,3 L21,9 L15,9\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div><div class=\\"css-994968-MuiStack-root\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-dtznkg-MuiTypography-root\\">Munich, Germany</p><div dir=\\"ltr\\" aria-roledescription=\\"carousel\\" class=\\"css-1ohr7kt\\"><div class=\\"react-slideshow-container\\"><button class=\\"MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium undefined nav disabled css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Previous Slide\\" data-type=\\"prev\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><polyline fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" points=\\"7 2 17 12 7 22\\" transform=\\"matrix(-1 0 0 1 24 0)\\"></polyline></svg></button><div class=\\"react-slideshow-wrapper slide \\"><div class=\\"images-wrap\\" style=\\"transform: translate(-0px); width: 0px;\\"><div data-index=\\"-1\\" aria-roledescription=\\"slide\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j94aqb-MuiPaper-root-MuiCard-root\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Avg. temperature</p><p class=\\"MuiTypography-root MuiTypography-body1 css-a4lwg8-MuiTypography-root\\"><span>-1.02°C</span></p><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Mon, Nov 29</p></div></div></div><div data-index=\\"0\\" class=\\"active\\" aria-roledescription=\\"slide\\" aria-hidden=\\"false\\" style=\\"width: 0px;\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j94aqb-MuiPaper-root-MuiCard-root\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Avg. temperature</p><p class=\\"MuiTypography-root MuiTypography-body1 css-a4lwg8-MuiTypography-root\\"><span>-1.02°C</span></p><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Mon, Nov 29</p></div></div></div><div data-index=\\"1\\" aria-roledescription=\\"slide\\" aria-hidden=\\"true\\" style=\\"width: 0px;\\"><div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j94aqb-MuiPaper-root-MuiCard-root\\"><div class=\\"MuiCardContent-root css-46bh2p-MuiCardContent-root\\"><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Avg. temperature</p><p class=\\"MuiTypography-root MuiTypography-body1 css-a4lwg8-MuiTypography-root\\"><span>-1.02°C</span></p><p class=\\"MuiTypography-root MuiTypography-body1 css-p09gn3-MuiTypography-root\\">Mon, Nov 29</p></div></div></div></div></div><button class=\\"MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium undefined nav disabled css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Next Slide\\" data-type=\\"next\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><polyline fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" points=\\"7 2 17 12 7 22\\"></polyline></svg></button></div></div><span readonly=\\"\\" class=\\"MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root css-10yivjf-MuiSlider-root\\"><span class=\\"MuiSlider-rail css-14pt78w-MuiSlider-rail\\"></span><span class=\\"MuiSlider-track css-1gv0vcd-MuiSlider-track\\"></span><span data-index=\\"0\\" class=\\"MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-eg0mwd-MuiSlider-thumb\\"><input tabindex=\\"-1\\" data-index=\\"0\\" aria-label=\\"Index\\" aria-orientation=\\"horizontal\\" aria-valuemax=\\"0\\" aria-valuemin=\\"0\\" aria-valuenow=\\"0\\" type=\\"range\\" min=\\"0\\" max=\\"0\\" style=\\"border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;\\" value=\\"0\\"></span></span></div><div class=\\"MuiBox-root css-929dze\\"><div class=\\"css-e53awj-MuiStack-root\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"slide to first card\\" data-mui-internal-clone-element=\\"true\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"none\\" stroke=\\"#000\\" stroke-width=\\"2\\" d=\\"M1,11 L12,2 L23,11 M15,23 L15,15 L15,15 L9,15 L9,23 M4,23 L4,9 M20,23 L20,9\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></div></div>"`;
